src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -752,42 +752,29 @@
        sort))
 
 (defn extract-fields
-  "Return all fields from a projection, if expand? true. If expand? false,
-   returns all fields except for expanded ones, which are returned as
-   [<identifier> k], where identifier is the component used to make the href.
-
-   Return nil for fields which are query-only? since these can't be projected
-   either."
-  [[name {:keys [expandable? field]}] entity expand?]
-  (let [href-key (case entity
-                       :catalogs :certname
-                       :factsets :certname
-                       :reports :hash
-                       nil)]
-    (if (or expand? (not expandable?))
-      [field name]
-      [href-key name])))
+  "Returns all fields from a projection.  Returns nil for fields which
+  are query-only? since these can't be projected either."
+  [[name {:keys [field]}] entity]
+  [field name])
```

Comment:
@ajroetker hmm, not sure yet, but I might prefer the destructuring since the names make it a bit clearer to me, though I might also be happy with `[[name info] entity] [(:field info) name]` if that's preferable.